### PR TITLE
Add BGP Next Hop field to flow message

### PR DESCRIPTION
**Pull request by simPod**

### CODE REVIEW

File: producer/producer_sf.go [hunk 1]
```diff
@@ -266,7 +266,7 @@ func SearchSFlowSamplesConfig(samples []interface{}, config *SFlowProducerConfig
 				flowMessage.DstNet = recordData.DstMaskLen
 			case sflow.ExtendedGateway:
 				ipNh = recordData.NextHop
-				flowMessage.NextHop = ipNh
+				flowMessage.BgpNextHop = ipNh
```
Comment (lspgn) [inline producer/producer_sf.go:269]: Why adding a distinct field? Are there cases where NextHop (from ExtendedRouter) and BgpNextHop (from ExtendedGateway) are different?

Comment (simPod) [reply producer/producer_sf.go:269]: Yup, that was interesting find actually.

https://www.cisco.com/en/US/technologies/tk648/tk362/technologies_white_paper09186a00800a3db9.html

image

Should be covered by the test where `bgp next hop ip` is `5.5.5.5` and next hop ip is `9.9.9.9`

**Comment by lspgn** [comment]
Since this seems to be available on both sFlow and NetFlow, can you add also the insertion for `producer_nf.go`?

But still wondering if it is useful in a production setup. Are there implementations providing the two fields?

What I want to avoid is adding many vendor/protocol-dependent fields that will be zero in 99% of the time (other OSS projects have a unique JSON schema per protocol).

**Comment by simPod** [comment]
> Are there implementations providing the two fields?

@lspgn actually, the reason I've found this field is that I started consuming NextHopIp field and the value made no sense. So I debugged the packets our Aristas were sending and found out there are two distinct Next Hop Ip fields, each meaning something else. Does that somehow answer your question?

I've added it to NF producer, though I have no way to test it in live environment as I work only with sflow.

**Comment by lspgn** [comment]
Thanks! This is helpful. Can you provide me with a small sFlow sample so I can test as well.
I'll check if the number 100 also makes sense.
Otherwise should be good.

**Comment by simPod** [comment]
I have transformed the sample I had into the `getSflowPacket()` test method. But sure, I'll try. In what form would you wish it to be provided? bytes payload?

Also, I have opened relevant PR here https://github.com/netsampler/goflow2/pull/12.

**Comment by lspgn** [comment]
Either full bytes payload or pcap

**Pull request merged**